### DISCUSSION TITLE
Simple vim plugin for Pallene

### DISCUSSION THREAD

**Discussion by JLPLabs** [pallene-lang/pallene#461] [Show and tell]
I just spent a couple of hours trying to learn how to build a simple vim plugin.
It is my first attempt. It is sure to be open to LOTS of improvement!

I'm using vim8, so plugins are really easy (in terms of just creating a few files in the 'right' place and vim reads them for free).

Here is the tree:

```
.
├── ftdetect
│   └── pallene.vim
├── ftplugin
│   └── pallene.vim
├── indent
│   └── pallene.vim
└── syntax
    └── pallene.vim

```
The zip file contains these files.
[pallene.zip](https://github.com/pallene-lang/pallene/files/6988629/pallene.zip)

```
NOTE  I do NOT know if/how this plugin and and .editorconfig do or do not cooperate.
      So far they seem to be co-existing "ok".
```

**Comment by hugomg** [comment]
Interesting. If this Vim plugin ends up working for you, we could create a separate repository for it. IIRC, the various plugin managers prefer when the plugin is it's own github repository.

I'm not very good at creating Vim plugins, so I can't give many suggestions about it. All I can say that I know that Teal has [a Vim plugin](https://github.com/teal-language/vim-teal). It's another typed dialect of Lua and their vim plugin may be a good source of inspiration.

**Reply by JLPLabs** [reply]
I use teal also. Good idea to check into its plugin!

**Reply by JLPLabs** [reply]
I made a few improvements (did look into teal, it helped, thanks!).
I put in my area: https://github.com/JLPLabs/pallene-vim

**Reply by hugomg** [reply]
Neat! I'll give it a test drive later.

About the automatic "end" indentation, I think there's a plugin for that called "endwise". Maybe it will simplify the work into the Pallene plugin, if we don't have to reimplement things that are also applicable to other languages (like Lua or Teal).
